component/loki/source/file/file.go

```diff
@@ -182,7 +182,6 @@ func (c *Component) Update(args component.Arguments) error {
 		}
 
 		c.reportSize(path, labels.String())
-		c.handler = make(loki.LogsReceiver)
```

Comment:
Each time we enter Update, the entryHandler will be stopped and c.handler will be closed. Don't we want to re-create it here?

Reply:
We probably shouldn't close the handler until the component leaves, let me take a look.

Reply:
If not, then we can remove it from the stop func. The test will show if we have a goroutine leak.

Reply:
Moved the close to defer on the run method. Which should work since its only created once.